evm_loader/program/src/instruction/collect_treasury.rs

```diff
@@ -0,0 +1,28 @@
+use crate::account::{Treasury, MainTreasury};
+use arrayref::{array_ref};
+use solana_program::{
+    account_info::AccountInfo, entrypoint::ProgramResult,
+    program_error::ProgramError,
+    pubkey::Pubkey,
+    sysvar::Sysvar,
+    rent::Rent,
+};
+
+pub fn process<'a>(program_id: &'a Pubkey, accounts: &'a [AccountInfo<'a>], instruction: &[u8]) -> ProgramResult {
+    solana_program::msg!("Instruction: Collect treasury");
+
+    let treasury_index = u32::from_le_bytes(*array_ref![instruction, 0, 4]);
+
+    let main_treasury = MainTreasury::from_account(&accounts[0])?;
+    let treasury = Treasury::from_account(program_id, treasury_index, &accounts[1])?;
+
+    let rent = Rent::get()?;
+    let minimal_balance_for_rent_exempt = rent.minimum_balance(treasury.data_len());
```

Comment:
Do we even have a case where `treasury.data_len() > 0`?

Reply:
We don't have that case at the moment, but I think we should be prepared for data to be in these accounts.